hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3RepositoryAutoConfiguration.java

```diff
@@ -83,11 +83,10 @@ public ClientConfiguration awsClientConfiguration() {
     public AmazonS3 amazonS3() {
         final AmazonS3ClientBuilder s3ClientBuilder = AmazonS3ClientBuilder.standard()
                 .withCredentials(awsCredentialsProvider()).withClientConfiguration(awsClientConfiguration());
-        if (!StringUtils.isEmpty(region)) {
+        if (!StringUtils.isEmpty(endpoint)) {
+            s3ClientBuilder.withEndpointConfiguration(new EndpointConfiguration(endpoint, ""));
```

Comment:
Shouldn't you set region as well if it is provided via property `aws.region`?

Reply:
It is not necessary as it goes

Reply:
Sorry, for my briefness. I was thinking more about other users/programmers who maybe want to use manually set S3 endpoint and use AWS's `SigV4 signing` (which may require the region to be set). What do you think? Will this work for MinIO as well?

```java
if (!StringUtils.isEmpty(endpoint)) {
   final String signingRegion = StringUtils.isEmpty(region) ? "" : region;
   s3ClientBuilder.withEndpointConfiguration(new EndpointConfiguration(endpoint, signingRegion));
} else if (!StringUtils.isEmpty(region)) {
   s3ClientBuilder.withRegion(region);
}
```

Reply:
Probably. I will try to build it and test tomorrow.